File: backend/EMS.Domain/Entities/Employee.cs
using System;

namespace EMS.Domain.Entities
{
    public class Employee
    {
        public Guid Id { get; set; }
        public Guid CompanyId { get; set; }
        public Company? Company { get; set; }
        public string EmployeeCode { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string? MiddleName { get; set; }
        public string LastName { get; set; } = null!;
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public string? Gender { get; set; }
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? PostalCode { get; set; }
        public string? Country { get; set; }
        public string? EmergencyContactName { get; set; }
        public string? EmergencyContactPhone { get; set; }
        public string? EmergencyContactRelation { get; set; }
        public DateTime JoinDate { get; set; }
        public Guid? DepartmentId { get; set; }
        public Guid? TeamId { get; set; }
        public Guid DesignationId { get; set; }
        public Guid? ManagerId { get; set; }
        public Guid OfficeLocationId { get; set; }
        public Guid? ProfilePhotoDocumentId { get; set; }
        public string? EmploymentStatus { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime? ExitDate { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime CreatedAtUtc { get; set; }
        public DateTime? UpdatedAtUtc { get; set; }

        public Domain.Entities.Department? Department { get; set; }
        public Domain.Entities.Team? Team { get; set; }
        public Domain.Entities.Designation? Designation { get; set; }
        public Domain.Entities.OfficeLocation? OfficeLocation { get; set; }
        public Domain.Entities.Employee? Manager { get; set; }
        public Domain.Entities.EmployeeDocument? ProfilePhotoDocument { get; set; }
    }
}

File: frontend/web/src/app/shared/models/employee.ts
export type EmploymentStatus = 'Active' | 'Inactive' | 'OnLeave' | 'Terminated'

export interface EmployeeAddress {
  addressLine1: string | null
  addressLine2: string | null
  city: string | null
  state: string | null
  postalCode: string | null
  country: string | null
}

export interface EmployeeEmergencyContact {
  name: string | null
  phone: string | null
  relation: string | null
}

export interface Employee {
  id: string
  employeeCode: string
  firstName: string
  middleName: string | null
  lastName: string
  fullName: string
  email: string | null
  phoneNumber: string | null
  dateOfBirth: string | null
  gender: string | null
  address: EmployeeAddress
  emergencyContact: EmployeeEmergencyContact
  joinDate: string
  exitDate: string | null
  departmentId: string | null
  departmentName: string | null
  teamId: string | null
  teamName: string | null
  designationId: string
  designationName: string | null
  managerId: string | null
  officeLocationId: string
  officeLocationName: string | null
  profilePhotoDocumentId: string | null
  employmentStatus: EmploymentStatus | null
  isActive: boolean
  isDeleted: boolean
  createdAtUtc: string
  updatedAtUtc: string | null
}

export interface EmployeeListFilters {
  page?: number
  pageSize?: number
  search?: string
  departmentId?: string
  teamId?: string
  designationId?: string
  officeLocationId?: string
  status?: EmploymentStatus
  joinDateFrom?: string
  joinDateTo?: string
}

/** Mirrors backend/EMS.Application/Features/Employees/Commands/CreateEmployeeCommand.cs. */
export interface EmployeeWriteInput {
  employeeCode: string
  firstName: string
  middleName?: string
  lastName: string
  email?: string
  phoneNumber?: string
  dateOfBirth?: string
  gender?: string
  address?: Partial<EmployeeAddress>
  emergencyContact?: Partial<EmployeeEmergencyContact>
  joinDate: string
  departmentId?: string
  teamId?: string
  designationId: string
  managerId?: string
  officeLocationId: string
  employmentStatus?: EmploymentStatus
}

export type CreateEmployeeInput = EmployeeWriteInput

export interface UpdateEmployeeInput extends EmployeeWriteInput {
  id: string
}

/** Mirrors backend UpdateEmployeeStatusCommand.cs. */
export interface UpdateEmployeeStatusInput {
  id: string
  status: EmploymentStatus
  exitDate?: string
  reason?: string
}
